skyportal/models/analysis.py

```diff
@@ -152,3 +187,196 @@ def authinfo(self):
     @authinfo.setter
     def authinfo(self, value):
         self._authinfo = value
+
+
+class AnalysisMixin:
+    def calc_hash(self):
+        md5_hash = hashlib.md5()
+        md5_hash.update(self._data.to_netcdf())
+        self.hash = md5_hash.hexdigest()
+
+    def load_data(self):
+        """
+        Load the associated analysis data from disk.
+        """
+        self._data = xr.load_dataset(self.filename)
+
+    def save_data(self):
+        """
+        Save the associated analysis data to disk.
+        """
+
+        # there's a default value but it is best to provide a full path in the config
+        root_folder = cfg.get('analysis_folder', 'analysis_data')
+
+        # the filename can have alphanumeric, underscores, + or -
+        self.check_path_string(self._unique_id)
+
+        # make sure to replace windows style slashes
+        subfolder = self._unique_id.replace("\\", "/")
+
+        filename = f'analysis_{self.id}.nc'
+
+        path = os.path.join(root_folder, subfolder)
+        if not os.path.exists(path):
+            os.makedirs(path)
+
+        full_name = os.path.join(path, filename)
+
+        if len(full_name) > MAX_FILEPATH_LENGTH:
+            raise ValueError(
+                f'Full path to file {full_name} is longer than {MAX_FILEPATH_LENGTH} characters.'
+            )
+
+        self._data.to_netcdf(full_name)
+        self.filename = full_name
+
+    def delete_data(self):
+        """
+        Delete the associated data from disk
+        """
+
+        if os.path.exists(self.filename):
+            os.remove(self.filename)
+
+    @staticmethod
+    def check_path_string(string, allow_slashes=False):
+        if allow_slashes:
+            reg = RE_SLASHES
+        else:
+            reg = RE_NO_SLASHES
+
+        if not reg.match(string):
+            raise ValueError(f'Illegal characters in string "{string}". ')
+
+    @hybrid_property
+    def data(self):
+        """Lazy load the data dictionary"""
+        if self._data is None:
+            self.load_data()
+        return self._data
+
+    _unique_id = sa.Column(
+        sa.String,
+        nullable=False,
+        unique=True,
+        default=lambda: str(uuid.uuid4()),
+        doc='Unique identifier for this analysis result.',
+    )
+
+    hash = sa.Column(
+        sa.String,
+        nullable=True,
+        unique=True,
+        doc='MD5sum hash of the data to be saved to file. Prevents duplications.',
+    )
+
+    show_parameters = sa.Column(
+        sa.Boolean,
+        default=False,
+        nullable=False,
+        doc="Whether to render the parameters of this analysis",
+    )
+
+    show_plots = sa.Column(
+        sa.Boolean,
+        default=False,
+        nullable=False,
+        doc="Whether to render the plots of this analysis",
+    )
+
+    show_corner = sa.Column(
+        sa.Boolean,
+        default=False,
+        nullable=False,
+        doc="Whether to render the corner plots of this analysis",
+    )
```

Comment:
Do we want to show/not show things on the individual object analysis level or on the analysis service level?

Reply:
Probably want it at the analysis level -- I can imagine wanting allow only certain groups to see some plots for some analysis and not for others.

Reply:
If I understand correctly, the way it is now, when you send a specific source to be analyzed you decide if you want to plot it as well. I'm not sure how that happens on a group level.